src/proto/single/ent.rs

```diff
@@ -67,13 +84,198 @@ impl JobBuilder {
+}
+
+impl Progress {
+    /// Create an instance of `ProgressUpdate` for the job updating its completion percentage.
+    ///
+    /// This will copy the [`desc`](Progress::desc) from the `Progress` (retrieved) over to `ProgressUpdate` (to be sent).
+    pub fn update_percent(&self, percent: u8) -> ProgressUpdate {
+        ProgressUpdate::builder(&self.jid)
+            .desc(self.desc.clone())
+            .percent(percent)
+            .build()
+    }
+
+    /// Create an instance of `ProgressUpdateBuilder` for the job.
+    pub fn update_builder(&self) -> ProgressUpdateBuilder {
+        ProgressUpdateBuilder::new(&self.jid)
+    }
+}
+
+/// Info on job execution progress (sent).
+///
+/// In Enterprise Faktory, a client executing a job can report on the execution
+/// progress, provided the job is trackable. A trackable job is one with "track":1
+/// specified in the custom data hash.
+#[derive(Debug, Clone, Serialize, Builder)]
+#[builder(
+    custom_constructor,
+    setter(into),
+    build_fn(name = "try_build", private)
+)]
+pub struct ProgressUpdate {
+    /// Id of the tracked job.
+    #[builder(setter(custom))]
+    pub jid: String,
+
+    /// Percentage of the job's completion.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[builder(default = "None")]
+    pub percent: Option<u8>,
+
+    /// Arbitrary description that may be useful to whoever is tracking the job's progress.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[builder(default = "None")]
+    pub desc: Option<String>,
+
+    /// Allows to extend the job's reservation, if more time is needed to execute it.
+    ///
+    /// Note that you cannot shorten the initial [reservation](struct.Job.html#structfield.reserve_for) via
+    /// specifying an instant that is sooner than the job's initial reservation deadline.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[builder(default = "None")]
+    pub reserve_until: Option<DateTime<Utc>>,
+}
+
+impl ProgressUpdate {
+    /// Create a new instance of `ProgressUpdateBuilder` with job ID already set.
+    ///
+    /// Equivalent to creating a [new](struct.ProgressUpdateBuilder.html#method.new)
+    /// `ProgressUpdateBuilder`.
+    pub fn builder(jid: impl Into<String>) -> ProgressUpdateBuilder {
+        ProgressUpdateBuilder::new(jid)
+    }
+}
+
+impl ProgressUpdateBuilder {
+    /// Builds an instance of ProgressUpdate.
+    pub fn build(&self) -> ProgressUpdate {
+        self.try_build()
+            .expect("All required fields have been set.")
+    }
+
+    /// Create a new instance of `JobBuilder`
+    pub fn new(jid: impl Into<String>) -> ProgressUpdateBuilder {
+        ProgressUpdateBuilder {
+            jid: Some(jid.into()),
+            ..ProgressUpdateBuilder::create_empty()
+        }
+    }
+}
+
+/// Create an instance of `ProgressUpdate` for the job specifying its completion percentage.
+pub fn set_progress(jid: impl Into<String>, percent: u8) -> ProgressUpdate {
```

Comment:
How about making this a method on `ProgressUpdate` that doesn't take `self`, so that it can be called as `ProgressUpdate::set(jid, percent)`?

Reply:
Yeah I like it better when it's in the `ProgressUpdate` namespace!